display_information/update-graphics/src/main/java/com/esri/samples/update_graphics/UpdateGraphicsSample.java

```diff
@@ -152,7 +153,7 @@ public void start(Stage stage) {
       selectedGraphic = new Graphic();
 
       // create a ArcGISMap with basemap light gray canvas
-      ArcGISMap map = new ArcGISMap(Basemap.Type.LIGHT_GRAY_CANVAS, 56.075844, -2.681572, 13);
+      map = new ArcGISMap(Basemap.Type.LIGHT_GRAY_CANVAS, 56.075844, -2.681572, 13);
 
       // enable buttons when map view is done loading
       map.addDoneLoadingListener(() -> {
```

Comment:
@colinanderson could you comment on this please? I think I was under the impression we would still need it so that there isn't potentially a reference lost for the addDoneLoadingListener, but perhaps it is just for cases when the loadable is called again within the done loading listener.

Reply:
>Just for my own understanding: we need map as a member variable even if we're not using that object within it's own DoneLoadingListener?

@Rachael-E @JonLavi 
Yes, the garbage collection issue is that the object  can get garbage collected because it is unreferenced and the done listener never runs. Referencing the loadable inside the done listener doesn't stop it being garbage collected - I assume it becomes a circular reference to a local variable and the variable isn't referenced anymore in the method so can be garbage collected. I'd have to do some revision to get the exact facts though!

Reply:
great, thanks Colin!